### PR TITLE
Alternate fix for import highlighting with instantiations

### PR DESCRIPTION
**Pull request by dbartol**
This is an alternate fix for https://github.com/github/vscode-codeql/pull/2792. It updates the heuristic for finding the end of an import directive. The new heuristic matches a zero-width string that is preceded by a character that could end the import directive (identifier, `>`) and is not followed by a character that would continue the import directive (`,`, `.`, `<`, `::`).

### CODE REVIEW

File: extensions/ql-vscode/syntaxes/ql.tmLanguage.yml [hunk 1]
```diff
@@ -661,29 +661,38 @@ repository:
...
-    # line breaks between the simple-id and the `.` or `::`. 
-    end: '(?#simple-id) (?!\s*(\.|\:\:))'
-    endCaptures:
-      '0':
-        name: entity.name.type.namespace.ql
+    # TextMate makes it tricky to tell whether an identifier that we encounter is part of the
+    # `import` directive or whether it's the first token of the next module-level declaration.
+    # To find the end of the import directive, we'll look for a zero-width match where the previous
+    # token is either an identifier (other than `import`) or a `>`, and the next token is not a `.`,
+    # `<`, `,`, or `::`. This works for nearly all real-world `import` directives, but it will end the
```
Comment (ginsbach) [inline extensions/ql-vscode/syntaxes/ql.tmLanguage.yml:685]: I think we also need '>' in that list of allowed next tokens. After all, we could import a nested instanitation
```ql
bindingset[this]
signature class TSig;

module M1<TSig T> {
  class type = T;
}

module M2 {
  import M1<M1<int>>
}
```

**Review by ginsbach — CHANGES_REQUESTED** [review]
Thanks a lot for looking into this! Apologies for taking a bit of time to get back to you. I have this one minor suggestion, aside from that it looks good to me.

File: extensions/ql-vscode/syntaxes/ql.tmLanguage.yml (hunk 1 above)
Comment (ginsbach) [reply extensions/ql-vscode/syntaxes/ql.tmLanguage.yml:685]: Actually, it seems there is no discernible difference, and even the compiled grammar stays the same if I add '>'.

**Review by ginsbach — APPROVED** [review]
I am approving this issue to signal that I'm happy with the changes, but there is no need to merge - I have already updated [my branch](https://github.com/github/vscode-codeql/pull/2792)

**Comment by dbartol** [comment]
Closing since this was already merged into the real PR

**Pull request closed without merging**